server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -901,7 +903,9 @@ protected Node(
             );
 
             List<HealthIndicatorService> serverHealthIndicatorServices = List.of(
-                new InstanceHasMasterHealthIndicatorService(clusterService)
+                new InstanceHasMasterHealthIndicatorService(clusterService),
+                new NoEligibleMasterNodesIndicator(clusterService.state()),
+                new LackOfQuorumIndicator(clusterService.state())
```

Comment:
I believe this executed only once during node start, so `NoEligibleMasterNodesIndicator` and `LackOfQuorumIndicator` might have outdated states by the time they are queried.

Reply:
Thank you! I've already refactored the indicators to accept `ClusterService`.